### DISCUSSION TITLE
Set specified filenames for non-merged, zipped PDFs using libreoffice?

### DISCUSSION THREAD

**Discussion by chrishughes20** [gotenberg/gotenberg#769] [Q&A]
Hello!

I am wondering if it's possible to somehow set a specific filename for PDF files that ultimately get zipped? I have cloned the project and am running with the log level set to debug. In the logs, I see the filename I want for each filename but then the output filename is changed to a random filename. Here is part of the log for one of my files:
`{"level":"debug","ts":1704735359.0625625,"logger":"api.formslibreofficeconvert.usrbinunoconv.stderr","msg":"Input file: /tmp/aa2cf6de-5a9c-47c6-8fe2-694c777b5798/visit-report_bart-simpson_2023-12-24.pdf","trace":"a1461428-daa0-47dc-adee-dace4c651afe"}
{"level":"debug","ts":1704735359.1313896,"logger":"api.formslibreofficeconvert.usrbinunoconv.stdout","msg":"Output file: file:///tmp/aa2cf6de-5a9c-47c6-8fe2-694c777b5798/f6d3268d-5a7d-4088-b65f-4eb87338a733.pdf","trace":"a1461428-daa0-47dc-adee-dace4c651afe"}`

The filename I would like to be used as the filename is `visit-report_bart-simpson_2023-12-24.pdf` rather than the output filename `f6d3268d-5a7d-4088-b65f-4eb87338a733.pdf`. 

Also, I'm a total Go newb and have been trying to play around with a cloned version of the project and have added a few `c.Logger().Debug("we are here")` and `fmt.Print("Inside of route")` inside of `pkg/modules/libreoffice/routes.go` but I don't see this output anywhere in the console or Docker logs after rebuilding and restarting the Docker container. Could this output be being saved to a log file or something somewhere or am I (likely) doing something wrong?

Thank you for your help!

**Comment by gulien** [comment]
Hello @chrishughes20 👋 

Indeed, with many files it make sense to keep the original file name. I’ll add this to my TODO.
